### ISSUE TITLE
JSON field mismatch

### ISSUE THREAD

**Issue by ChayaSt** [leeping/geomeTRIC#22]
I'm trying to run geometric with the JSON interface. Here is the input JSON:
```
 qc_schema_input = {
  7         "schema_name": "QC_JSON",
  8         "schema_version": 0,
  9         "driver": "gradient",
 10         "molecule": {
 11             "geometry": [
 12                 0.31914281845092773,
 13                 -1.093637466430664,
 14                 -1.5644147396087646,
 15                 0.09283685684204102,
 16                -0.7512494325637817,
 17                 -0.10052239894866943,
 18                 -0.09279406070709229,
 19                 0.7513599395751953,
 20                 0.1004934310913086,
 21                 -0.290915846824646,
 22                 1.0967583656311035,
 23                 1.5677818059921265,
 24                 -0.5198796391487122,
 25                 -0.7551677227020264,
 26                 -2.180879592895508,
 27                 1.2282583713531494,
 28                -0.6067847013473511,
 29                 -1.931321620941162,
 30                 0.43323153257369995,
 31                 -2.1737723350524902,
 32                 -1.7016046047210693,
 33                 0.9486593008041382,
 34                 -1.1027858257293701,
 35                 0.48745453357696533,
 36                 -0.7917245626449585,
 37                 -1.287994146347046,
 38                 0.26173627376556396,
 39                 -0.960730254650116,
 40                 1.099387764930725,
 41                 -0.47136321663856506,
 42                 0.7833671569824219,
 43                 1.2891098260879517,
 44                 -0.28042128682136536,
 45                 -0.42176103591918945,
 46                 2.176612377166748,
 47                 1.6888495683670044,
 48                 0.5750753879547119,
 49                 0.7906937599182129,
 50                 2.16329026222229,
 51                 -1.1788761615753174,
 52                 0.5997258424758911,
 53                 1.971407413482666
 54             ],  
 55             "symbols": [
 56                 "C",
 57                 "C",
 58                 "C",
 59                 "C",
 60                 "H",
 61                 "H",
 62                 "H",
 63                 "H",
 64                 "H",
 65                 "H",
 66                 "H",
 67                 "H",
 68                 "H",
 69                 "H"
 70             ]
 71         },
 72         "driver": "gradient",
 73         "model": {"method": "hf", "basis": "sto-3g"},
 74         "keywords": {},
 75         "program": "rdkit"
 76     }
 77 
 78 in_json_dict = {
 79         "schema_name": "qc_schema_optimization_input",
 80         "schema_version": 1,
 81         "keywords": {
 82             "coordsys": "tric",
 83             "maxiter": 100
 84         },
 85         "input_specification": qc_schema_input
 86     }
```

When I specify the psi4 program I get the following error:
```
Traceback (most recent call last):
  File "run_geometric_json.py", line 88, in <module>
    out_json_dict_2 = geometric.run_json.geometric_run_json(in_json_dict_2)
  File "/home/chayas/anaconda3/envs/psi4-1.2rc2/lib/python3.6/site-packages/geometric-0.8.2+42.g5f5690c-py3.6.egg/geometric/run_json.py", line 102, in geometric_run_json
    geometric.optimize.Optimize(coords, M, IC, engine, dirname, params)
  File "/home/chayas/anaconda3/envs/psi4-1.2rc2/lib/python3.6/site-packages/geometric-0.8.2+42.g5f5690c-py3.6.egg/geometric/optimize.py", line 886, in Optimize
    E, gradx = engine.calc(coords, dirname)
  File "/home/chayas/anaconda3/envs/psi4-1.2rc2/lib/python3.6/site-packages/geometric-0.8.2+42.g5f5690c-py3.6.egg/geometric/engine.py", line 703, in calc
    return self.calc_new(coords, dirname)
  File "/home/chayas/anaconda3/envs/psi4-1.2rc2/lib/python3.6/site-packages/geometric-0.8.2+42.g5f5690c-py3.6.egg/geometric/engine.py", line 698, in calc_new
    gradient = np.array(ret["return_value"])
KeyError: 'return_value'
```
This is what the JSON output looks like:
```
{'schema_name': 'QC_JSON', 
'schema_version': 0, 
'molecule': {
    'geometry': [0.31914281845092773, -1.093637466430664, -1.5644147396087646, 0.09283685684204102, -0.7512494325637817, -0.10052239894866943, -0.09279406070709229, 0.7513599395751953, 0.1004934310913086, -0.290915846824646, 1.0967583656311035, 1.5677818059921265, -0.5198796391487122, -0.7551677227020264, -2.180879592895508, 1.2282583713531494, -0.6067847013473511, -1.9313216209411623, 0.43323153257369995, -2.1737723350524902, -1.7016046047210693, 0.9486593008041382, -1.1027858257293701, 0.48745453357696533, -0.7917245626449585, -1.287994146347046, 0.26173627376556396, -0.960730254650116, 1.099387764930725, -0.47136321663856506, 0.7833671569824219, 1.2891098260879517, -0.28042128682136536, -0.42176103591918945, 2.176612377166748, 1.6888495683670044, 0.5750753879547119, 0.7906937599182129, 2.16329026222229, -1.1788761615753174, 0.5997258424758911, 1.971407413482666], 
    'symbols': ['C', 'C', 'C', 'C', 'H', 'H', 'H', 'H', 'H', 'H', 'H', 'H', 'H', 'H'], 
    'fix_orientation': True, 
    'fix_com': True}, 
'driver': 'gradient', 
'model': {
    'method': 'hf', 
    'basis': 'sto-3g'}, 
'keywords': {}, 
'program': 'psi4', 
'memory': 1932735283, 
'success': True, 
'raw_output': None, 
'provenance': {
    'creator': 'Psi4', 
    'version': '1.2rc2', 
    'routine': 'psi4.json.run_json', 
    'cpu': 'Intel(R) Xeon(R) CPU E5-2697 v4 @ 2.30GHz',
    'hostname': 'ls01', 
    'username': 'chayas'}, 
'psi4:qcvars': {
    'CURRENT DIPOLE X': -0.00589204557899569, 
    'CURRENT DIPOLE Y': 0.011250413621672823, 
    'CURRENT DIPOLE Z': -0.0008021024951966419,
    'CURRENT ENERGY': -144.34014427128707,  
    'CURRENT REFERENCE ENERGY': -144.34014427128707, 
    'HF TOTAL ENERGY': -144.3401442712871, 
    'NUCLEAR REPULSION ENERGY': 247.8811587307185, 
    'ONE-ELECTRON ENERGY': -664.7301173229439, 
    'PCM POLARIZATION ENERGY': 0.0, 
    'SCF DIPOLE X': -0.00589204557899569, 
    'SCF DIPOLE Y': 0.011250413621672823, 
    'SCF DIPOLE Z': -0.0008021024951966419, 
    'SCF ITERATION ENERGY': -144.34014427128707, 
    'SCF ITERATIONS': 7.0, 
    'SCF N ITERS': 7.0, 
    'SCF TOTAL ENERGY': -144.34014427128707, 
    'TWO-ELECTRON ENERGY': 272.50881432093826}, 
'return_result': [-0.3060768484680957, 0.7234438982361766, 1.9318407036014387, 0.03981452663663909, 1.342284843172246, -1.3307353583014359, -0.0008277640816439502, -1.3318412477589523, 1.3322453383233177, 0.2897062618702261, -0.6945841005920805, -1.938405625452312, 1.6460046024922417, -0.638461302007268, 1.3291451533205638, -1.82067129743742, -0.9347701663391986, 0.840745611076298, -0.2449448643818894, 2.1740910507365836, 0.3879621567637566, -1.6520729336086257, 0.7597279386868547, -1.2227520596303343, 1.7147190773785341, 1.115249416804866, -0.7891705715095088, 1.6785031200297202, -0.7507788731943492, 1.1948646854302098, -1.6945908937996765, -1.1188348411371976, 0.8223806797482305, 0.27721094637951815, -2.178341043591783, -0.3620843481047197, -1.7016269610301038, 0.5766870283586947, -1.2884448017390449, 1.7748530280204144, 0.956127398624706, -0.9075915635255932], 
'properties': {
    'calcinfo_nbasis': 30, 
    'calcinfo_nmo': 30, 
    'calcinfo_nalpha': 17, 
    'calcinfo_nbeta': 17, 
    'calcinfo_natom': 14, 
    'scf_one_electron_energy': -664.7301173229439, 
    'scf_two_electron_energy': 272.50881432093826, 
    'nuclear_repulsion_energy': 247.8811587307185, 
    'scf_dipole_moment': [-0.00589204557899569, 0.011250413621672823, -0.0008021024951966419], 
    'scf_iterations': 7, 
    'scf_total_energy': -144.34014427128707, 
    'return_energy': -144.34014427128707}, 
'wall_time': 2.1866307258605957}
```
When I change [this line](https://github.com/leeping/geomeTRIC/blob/master/geometric/engine.py#L697) to `gradient = np.array(ret["return_result"])` it works, however if I use `"program": "rdkit"`, I get the same error but this time the `KeyError` is for `return_results`.

**Comment by dgasmith** [comment]
As a note, probably an issue on my end. Will try to get to it this weekend.

**Comment by leeping** [comment]
Thanks @dgasmith 👍

**Comment by dgasmith** [comment]
Yup, found it. I need a few days for versions to correctly percolate through the system (should hopefully be the last time as the velocity of schema change is slowing). Will update this on Monday.

**Comment by dgasmith** [comment]
This was fixed in #23, please close this issue when you have a moment.
